sql/sql_parse.cc

```diff
@@ -6390,6 +6456,25 @@ static bool execute_sqlcom_select(THD *thd, TABLE_LIST *all_tables)
           return 1;
         if (vtmd.exists && vtmd.setup_select(thd))
           return 1;
+
+        if (thd->variables.vers_ident_mode == VERS_IDENT_MODE_CURRENT &&
+            table->vers_conditions == FOR_SYSTEM_TIME_AS_OF)
+        {
+          Dynamic_array<String_pair> map;
+          LEX_STRING db = {table->db, table->db_length};
+          LEX_STRING current_name = {table->alias, strlen(table->alias)};
+          LEX_STRING archive_name = {table->table_name, table->table_name_length};
+          if (get_fields_mapping(thd, db, current_name, archive_name, map))
+            return 1;
+
+          for (Item *item= thd->stmt_arena->free_list; item; item= item->next)
```

Comment:
Why you iterating `free_list`? It's list of free items, how they can be used?

Reply:
This is not a list of freed items but a list of items to free.
Every item got put on this list in `Item::Item()`.

Reply:
Mm... I guess such semantics anyway does not allow to use it.

Reply:
Is `stmt_arena` correct under SP, prepared, derived?